platform-hub-web/src/app/kubernetes-tokens/kubernetes-tokens-form.html

```diff
@@ -0,0 +1,83 @@
+<div class="kubernetes-tokens-form">
+  <md-toolbar md-scroll-shrink>
+    <div class="md-toolbar-tools">
+      <h3>
+        <span ng-if="$ctrl.isNew">New Token</span>
+        <span ng-if="!$ctrl.isNew">Edit "{{$ctrl.tokenData.cluster}}" token</span>
+        for {{$ctrl.user.name}} ({{$ctrl.user.email}})
+      </h3>
+    </div>
+  </md-toolbar>
+
+  <loading-indicator loading="$ctrl.loading || $ctrl.saving"></loading-indicator>
+
+  <md-content layout-padding ng-if="!$ctrl.loading">
+
+    <div flex-sm="100" flex-gt-sm="90" flex-gt-md="70" flex-gt-lg="50" class="md-whiteframe-z2">
+      <md-content layout-padding>
+        <form name="$ctrl.kubernetesTokenForm" role="form" ng-submit="$ctrl.createOrUpdate()">
+
+          <md-input-container class="md-block">
+            <label for="cluster">Cluster:</label>
+            <md-select ng-model="$ctrl.tokenData.cluster" 
+              name="cluster"
+              required
+              placeholder="Select the cluster"
+              ng-disabled="!$ctrl.isNew && $ctrl.tokenData.cluster"
+              >
+              <md-option
+                ng-repeat="o in $ctrl.clusters"
+                value="{{o.id}}">
+                {{o.name}}
+              </md-option>
+            </md-select>
+            <div ng-messages="$ctrl.kubernetesTokenForm.cluster.$error">
+              <div ng-message="required">This is required.</div>
+            </div>
+          </md-input-container>
+
+          <md-input-container class="md-block" ng-if="!$ctrl.isNew">
+            <label for="token">Token:</label>
+            <input
+              name="token"
+              ng-model="$ctrl.tokenData.token"
+              ng-disabled="$ctrl.tokenData.token"
+              required
+              placeholder="Kubernetes access token"
+              autofocus>
+            <div ng-messages="$ctrl.kubernetesTokenForm.token.$error">
+              <div ng-message="required">This is required.</div>
+            </div>
+          </md-input-container>
+
+          <md-input-container class="md-block">
+            <label for="groups">Groups:</label>
+            <input
+              name="groups"
+              ng-model="$ctrl.tokenData.groups"
+              required
+              placeholder="Comma separated list of groups.">
+            <div ng-messages="$ctrl.kubernetesTokenForm.groups.$error">
+              <div ng-message="required">Groups are required.</div>
+            </div>
+          </md-input-container>
```

Comment:
(maybe not worth it) We could add some HTML5 validation here (using a regex) to check the text entered matches a comma-separated list. Or maybe it's fine to just enforce this on the backend?